io/net/virt-net/network_virtualization.py

```diff
@@ -189,10 +187,41 @@ def rsct_service_start(self):
         if "inoperative" in output:
             self.cancel("Failed to start the rsct and rsct_rm services")
 
+    def install_packages(self):
+        '''
+        Install necessary packages
+        '''
+        smm = SoftwareManager()
+        detected_distro = distro.detect()
+        self.log.info("Test is running on %s", detected_distro.name)
+        if not smm.check_installed("ksh") and not smm.install("ksh"):
+            self.error('ksh is needed for the test to be run')
+        if detected_distro.name == "Ubuntu":
+            if not smm.check_installed("python-paramiko") and not \
+                    smm.install("python-paramiko"):
+                self.error('python-paramiko is needed for the test to be run')
+            ubuntu_url = self.params.get('ubuntu_url', default=None)
+            debs = self.params.get('debs', default=None)
+            for deb in debs:
+                deb_url = os.path.join(ubuntu_url, deb)
+                deb_install = self.fetch_asset(deb_url, expire='7d')
+                shutil.copy(deb_install, self.workdir)
+                process.system("dpkg -i %s/%s" % (self.workdir, deb),
+                               ignore_status=True, sudo=True)
+        else:
+            url = self.params.get('url', default=None)
```

Comment:
here also please if url is None , it need to cancel the test

Reply:
sure